years_to_hours.rb

```diff
@@ -0,0 +1,55 @@
+def run()
+    puts("Welcome! In this app, you'll be converting:
+        \r Years to hours
+        \r Decades to minutes
+        \r Your age to seconds\n")
+    puts(year_hour())
+    puts(decade_minute())
+    puts(age_second())
+    puts("You have reached the end of the program. Bye :)")
+end
+
+def input_handling(what_to_ask)
+    while true do
+        input = gets.chomp
+        # used Regex to check if it's a whole number
+        unless input.match(/[^0-9]/)
+            return input.to_i
+        else
+            #                                        what_to_ask is plural, this makes it singular
+            print "\nThat's not an integer.\nPlease insert a whole number (#{what_to_ask[..-2]}): "
+        end
+    end
+end
+
+def ask_num(what_to_ask)
+
+    if what_to_ask == "ages"
+        #                             age inseted of ages
+        print "\nPlease enter your #{what_to_ask[..-2]}: "
```

Comment:
How would you feel about handling this by passing in the singular ("year" instead of "years", "age" instead of "ages", etc.)? Then you wouldn't have to get any substrings.